### PR TITLE
fix: eta display in GTK client details dialog

### PR DESCRIPTION
**Pull request by ckerr**
Fix broken ETA formatting in the GTK client's display dialog.

### CODE REVIEW

**Comment by Coeur** [comment]
I know it's not the topic, but I have no other way to contact you:
Can you kindly look at #4222 and eventually rollback #4214 before the rest? Thanks

File: gtk/DetailsDialog.cc [hunk 1]
```diff
@@ -805,13 +805,13 @@ void DetailsDialog::Impl::refreshInfo(std::vector<tr_torrent*> const& torrents)
         {
             str = mixed;
         }
-        else if (baseline < 0)
+        else if (baseline == TR_ETA_NOT_AVAIL || baseline == TR_ETA_UNKNOWN)
```
Comment (Coeur) [inline gtk/DetailsDialog.cc:808]: I prefer `baseline < 0`: we don't want to show any negative value anyway.

**Review by Coeur — CHANGES_REQUESTED** [review]

File: gtk/DetailsDialog.cc (hunk 1 above)
Comment (ckerr) [reply gtk/DetailsDialog.cc:808]: There are no negative values anymore; you changed the type to a time_t :smile_cat:

Comment (Coeur) [reply gtk/DetailsDialog.cc:808]: I'm reading https://en.wikipedia.org/wiki/Year_2038_problem now.
While most systems went from a **signed** 32 bit integer to a **signed** 64 bit integer to fix the Year 2038 problem, some rare systems may have quirkly adopted an unsigned 32 bit integer to buy some time.

Comment (ckerr) [reply gtk/DetailsDialog.cc:808]: You're right!

Comment (ckerr) [reply gtk/DetailsDialog.cc]: Updated 5ee59a4dc3ed578e2de1f355769d96fd20dc3d40

**Review by Coeur — APPROVED** [review]

**Comment by Coeur** [comment]
Thanks. I hope we can get a beta 2, now. :)

**Comment by ckerr** [comment]
> Thanks. I hope we can get a beta 2, now. :)

Still need to fix that settings bug in the GTK client. I've been working on other things because I'm not sure what the Right Fix is for it. But yeah, maybe that's not a blocker for a beta.

**Comment by Coeur** [comment]
< joke >
revert #4053: no MACROs, no bugs in the GTK client. :D
</ joke >

**Comment by ckerr** [comment]
The root problem is that client settings are intermixed into the same bag as the session settings. There's another issue about this where running transmission-qt in remote mode picks up the daemon's settings and reuses them the next time you run transmission-qt in local mode.

We probably ought to make `settings.json` more structured but that would be a pretty major change so I've been trying to decide what the least harmful fix is.

**Comment by Coeur** [comment]
Well, it's the last known issue to resolve, so I'm sure you'll figure out something in the coming days.
But if all it takes is to implement back `tr_sessionGetDefaultSettings` and `tr_session::setImpl` the verbose way, then I vote to re-add back a hundred of lines to avoid the regression.

**Comment by mikedld** [comment]
> The root problem is that client settings are intermixed into the same bag as the session settings

Working on it, see [wip/gtk-gio-settings](https://github.com/transmission/transmission/compare/wip/gtk-gio-settings).

**Comment by GaryElshaw** [comment]
> Thanks. I hope we can get a beta 2, now. :)

Out of curiosity, is there a plan to do some 3.0 vs 4.0 performance testing before 4.0 final? I'm particularly curious about basic stuff:

Does the client have better connectivity with:
* Peers
* Trackers

How are we doing with upload and download speeds compared w/ 3.0?

How are we tracking when it comes to mitigating corruption in the download process from 3.0 to 4.0?

**Pull request merged**